### PR TITLE
Fix minor test wait bug

### PR DESCRIPTION
**Pull request by zivkovicmilos**
# Description

This PR fixes a minor bug in `TestGetTxs`, where there wasn't a proper event wait before checking the state of the TxPool, which would cause a race when asserting. 

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [ ] I have tested this code manually

# Additional Comments

Fixes EDGE-404

### CODE REVIEW

File: txpool/txpool_test.go [hunk 1]
```diff
@@ -1897,14 +1905,24 @@ func TestGetTxs(t *testing.T) {
...
 			ctx, cancelFn := context.WithTimeout(context.Background(), time.Second*10)
 			defer cancelFn()
 
-			// All txns should get added
-			assert.Len(t, waitForEvents(ctx, subscription, totalTx), totalTx)
+			// Wait for promoted transactions
+			assert.Len(t, waitForEvents(ctx, promoteSubscription, totalPromotedTx), totalPromotedTx)
+
+			// Wait for enqueued transactions, if any are present
+			totalEnqueuedTx := totalPromotedTx - len(test.allTxs)
```
Comment (Kourin1996) [inline txpool/txpool_test.go]: Nicpick, `totalPromotedTx` is just an expected value, right?

**Review by Kourin1996 — APPROVED** [review]
Looks good to me. I've left small comments.

File: txpool/txpool_test.go (hunk 1 above)
Comment (zivkovicmilos) [reply txpool/txpool_test.go]: Yes, it's the total number of transactions that should be promoted within the test case, so our `waitForEvents` call knows for how many events it should wait for

Comment (zivkovicmilos) [reply txpool/txpool_test.go]: The transaction can either be promoted or enqueued in this specific test, so there is no need to keep track of enqueued transactions when we already keep track of promoted ones - if a transaction is not promoted, then it is enqueued

**Review by dbrajovic — APPROVED** [review]

File: txpool/txpool_test.go (hunk 1 above)
Comment (zivkovicmilos) [reply txpool/txpool_test.go]: Fixed, in all places 👍 

[ccb1564](https://github.com/0xPolygon/polygon-edge/pull/421/commits/ccb1564169770421faa05b77fa9eb47abc4060a8)

**Pull request merged**